docs/resources/server/loottables/index.md

```diff
@@ -0,0 +1,437 @@
+          "conditions": [
+            // entry-wide loot conditions
+          ],
+          "functions": [
+            // entry-wide loot functions
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+## Rolling a Loot Table
+
+To roll a loot table, we need two things: the loot table itself, and a loot context.
+
+Let's start with getting the loot table itself. We can obtain a loot table using `level.getServer().reloadableRegistries().getLootTable(lootTableId)`. As the loot data is only available through the server, this logic must run on a [logical server][sides], not a logical client.
+
+:::tip
+Minecraft's built-in loot table IDs can be found in the `BuiltInLootTables` class. Block loot tables can be obtained through `BlockBehaviour#getLootTable`, and entity loot tables can be obtained through `EntityType#getDefaultLootTable` or `Entity#getLootTable`.
+:::
+
+Now that we have a loot table, let's build our parameter set. We begin by creating an instance of `LootParams.Builder`:
+
+```java
+// Make sure that you are on a server, otherwise the cast will fail.
+LootParams.Builder builder = new LootParams.Builder((ServerLevel) level);
+```
+
+We can then add loot context parameters, like so:
+
+```java
+// Use whatever context parameters and values you need. Vanilla parameters can be found in LootContextParams.
+builder.withParameter(LootContextParams.ORIGIN, position);
+// This variant can accept null as the value, in which case an existing value for that parameter will be removed.
+builder.withOptionalParameter(LootContextParams.ORIGIN, null);
+// Add a dynamic drop.
+builder.withDynamicDrop(ResourceLocation.fromNamespaceAndPath("examplemod", "example_dynamic_drop"), stack -> {
+    // some logic here
+});
+// Set our luck value. Assumes that a player is available. Contexts without a player should use 0 here.
+builder.withLuck(player.getLuck());
+```
+
+Finally, we can create the `LootParams` from the builder and use them to roll the loot table:
+
+```java
+// Specify a loot context param set here if you want.
+LootParams params = builder.create(LootContextParamSet.EMPTY);
+// Get the loot table.
+LootTable table = level.getServer().reloadableRegistries().getLootTable(location);
+// Actually roll the loot table.
+List<ItemStack> list = table.getRandomItems(params);
+// Use this instead if you are rolling the loot table for container contents, e.g. loot chests.
+// This method takes care of properly splitting the loot items across the container.
+List<ItemStack> containerList = table.fill(container, params, someSeed);
+```
+
+:::danger
+`LootTable` additionally exposes a method named `#getRandomItemsRaw`. Unlike the various `#getRandomItems` variants, `#getRandomItemsRaw` method will not apply [global loot modifiers][glm]. Use this method only if you know what you are doing.
+:::
+
+## Datagen
+
+Loot tables can be [datagenned][datagen] by subclassing `LootTableProvider` and providing a list of `LootTableSubProvider` in the constructor:
```

Comment:
You don't need to subclass `LootTableProvider` since everything is passed as a constructor parameter.